src/components/preference-item/preference-item.component.ts

```diff
@@ -0,0 +1,24 @@
+import { Component, Input } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+
+@Component({
+  selector: 'app-preference-item',
+  templateUrl: './preference-item.component.html',
+  styleUrls: ['./preference-item.component.scss'],
+})
+export class PreferenceItemComponent<T> {
```

Comment:
this generic seems pointless as it isn't set anywhere

Reply:
I think it would be less error prone to use generics (even though it isn't set anywhere) than making the type of `subject` to `BehaviorSubject<any>` or `BehaviorSubject<number | boolean | string>`.

```ts
subject: BehaviorSubject<T>;

// compile-time errors, which prevent runtime errors
subject.next('some_string');
subject.next(42);
```

```ts
subject: BehaviorSubject<any>;

// no compile-time errors, but potential runtime errors
subject.next('some_string');
subject.next(42);
```